yb-voyager/cmd/importData.go

```diff
@@ -1547,3 +1553,72 @@ func cleanStoredErrors(errorHandler importdata.ImportDataErrorHandler, tasks []*
 	}
 	return nil
 }
+
+func BuildCallhomeYBClusterMetrics() (callhome.YBClusterMetrics, error) {
+	yb, ok := tdb.(*tgtdb.TargetYugabyteDB)
+	if !ok {
+		utils.ErrExit("importData: expected tdb to be of type TargetYugabyteDB, got: %T", tdb)
+	}
+
+	clusterMetrics, err := yb.GetClusterMetrics()
+	if err != nil {
+		return callhome.YBClusterMetrics{}, err
+	}
+
+	now := time.Now().UTC()
+	nodes := make([]callhome.NodeMetric, 0)
+	var totalCpuPct, totalMemPct float64
+	for _, nodeMetrics := range clusterMetrics {
+		// in case of err value will be -1
+		cpuPct, err := nodeMetrics.CPUPercent()
+		if err != nil {
+			log.Warnf("callhome: error getting CPU percent for node %s: %v", nodeMetrics.UUID, err)
+		}
+		memPct, err := nodeMetrics.MemPercent()
+		if err != nil {
+			log.Warnf("callhome: error getting Mem percent for node %s: %v", nodeMetrics.UUID, err)
+		}
+
+		// in case of err value will be -1
+		memoryFree, err := nodeMetrics.GetMemoryFree()
+		if err != nil {
+			log.Warnf("callhome: error getting Memory Free for node %s: %v", nodeMetrics.UUID, err)
+		}
+		memoryAvailable, err := nodeMetrics.GetMemoryAvailable()
+		if err != nil {
+			log.Warnf("callhome: error getting Memory Available for node %s: %v", nodeMetrics.UUID, err)
+		}
+		memoryTotal, err := nodeMetrics.GetMemoryTotal()
+		if err != nil {
+			log.Warnf("callhome: error getting Memory Total for node %s: %v", nodeMetrics.UUID, err)
+		}
+
+		nodes = append(nodes, callhome.NodeMetric{
+			UUID:                   nodeMetrics.UUID,
+			TotalCPUPct:            cpuPct,
+			TserverMemSoftLimitPct: memPct,
+			MemoryFree:             memoryFree,
+			MemoryAvailable:        memoryAvailable,
+			MemoryTotal:            memoryTotal,
+			Status:                 nodeMetrics.Status,
+			Error:                  nodeMetrics.Error,
+		})
+
+		totalCpuPct += cpuPct
+		totalMemPct += memPct
+	}
+
+	if len(nodes) == 0 {
+		log.Error("callhome: no nodes found in cluster metrics")
+		return callhome.YBClusterMetrics{}, nil
+	}
+
+	avgCpuPct := totalCpuPct / float64(len(nodes))
+	avgMemPct := totalMemPct / float64(len(nodes))
+	return callhome.YBClusterMetrics{
+		Timestamp:                 now,
+		CPUAvgPct:                 avgCpuPct,
+		TserverAvgMemSoftLimitPct: avgMemPct,
```

Comment:
nit: this might not be very useful. 
avg cpu is interesting because it helps us understand whether on the whole, we are sufficiently utilising the cluster. 
I can't think of how avg-tservermemconsumption may be useful.

(no harm in keeping it ofc)

Reply:
Will Total Memory be meaningful/useful or NOT?